api/k8s/v1beta4/k8s_helpers.go

```diff
@@ -47,3 +47,62 @@ func (s *API) WaitForCluster(req *WaitForClusterRequest) (*Cluster, error) {
 	}
 	return cluster.(*Cluster), nil
 }
+
+// WaitForClusterNodesReadyRequest is used by WaitForClusterNodesReady method.
+type WaitForClusterNodesReadyRequest struct {
+	ClusterID string
+	Region    scw.Region
+	Timeout   time.Duration
+}
+
+// WaitForClusterNodesReady waits for the nodes of a cluster to be ready
+func (s *API) WaitForClusterNodesReady(req *WaitForClusterNodesReadyRequest) error {
+	pools, err := s.ListPools(&ListPoolsRequest{
+		ClusterID: req.ClusterID,
+		Region:    req.Region,
+	})
+	if err != nil {
+		return err
+	}
+
+	for _, pool := range pools.Pools {
+		err = s.WaitForPoolNodesReady(&WaitForPoolNodesReadyRequest{
+			PoolID:  pool.ID,
+			Timeout: req.Timeout,
+		})
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// WaitForPoolNodesReadyRequest is used by WaitForPoolNodesReady method.
+type WaitForPoolNodesReadyRequest struct {
+	PoolID  string
+	Timeout time.Duration
```

Comment:
Add the region and the same comment as above with the timeout

Reply:
And then you can do the same as here https://github.com/scaleway/scaleway-sdk-go/blob/047c88bb15c429f5061585865410530b78e13d15/api/k8s/v1beta4/k8s_sdk.go#L988

Reply:
Region added